src/Provider/TaxRatePercentageProvider.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\InvoicingPlugin\Provider;
+
+use Doctrine\Common\Collections\Collection;
+use Sylius\Component\Core\Model\AdjustmentInterface;
+use Sylius\Component\Order\Model\AdjustableInterface;
+use Sylius\InvoicingPlugin\Exception\MoreThanOneTaxAdjustment;
+use Webmozart\Assert\Assert;
+
+final class TaxRatePercentageProvider implements TaxRatePercentageProviderInterface
+{
+    public function provideFromAdjustable(AdjustableInterface $adjustable): ?string
+    {
+        /** @var Collection|AdjustmentInterface[] $taxAdjustments */
+        $taxAdjustments = $adjustable->getAdjustments(AdjustmentInterface::TAX_ADJUSTMENT);
+
+        if (count($taxAdjustments) > 1) {
+            throw MoreThanOneTaxAdjustment::occur();
+        }
+
+        if ($taxAdjustments->isEmpty()) {
+            return null;
```

Comment:
Is there any reason why we throw an exception in one undesired situation (more tax adjustments than one), but return a null in another (no tax adjustments at all)?

Reply:
no tax adjustment seems like totally ok situation. I'm considering if two tax adjustments are such an edge case, but I'm fine with this decision

Reply:
No tax adjustment is the normal case, but more than 2 tax adjustments is the case not supported by default by Sylius, so this is the reason why this exception is thrown here. What is more, there is a similar behaviour in RefundPlugin.

Reply:
Doubled taxation is not a particularly not supported feature of the main Sylius. We already support both included and excluded taxes on order items. Therefore it may happen. Nonetheless, I'm still fine with it for the first iteration. Especially, as RefundPlugin works in similar fashion